src/main/java/logisticspipes/compat/ModularUIHelper.java

```diff
@@ -0,0 +1,30 @@
+package logisticspipes.compat;
+
+import java.util.function.Supplier;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.world.World;
+
+import com.cleanroommc.modularui.api.drawable.IKey;
+import com.cleanroommc.modularui.drawable.UITexture;
+import com.cleanroommc.modularui.factory.TileEntityGuiFactory;
+
+import logisticspipes.LogisticsPipes;
+import logisticspipes.pipes.basic.CoreUnroutedPipe;
+
+public class ModularUIHelper {
+
+    public static final UITexture BACKGROUND_TEXTURE = UITexture.builder()
+            .location(LogisticsPipes.rl("textures/gui/GuiBackground.png")).imageSize(45, 45).adaptable(15).build();
+
+    public static void openPipeUI(EntityPlayer player, CoreUnroutedPipe pipe) {
+        World world = pipe.getWorld();
+        if (world == null || world.isRemote) return;
+
+        TileEntityGuiFactory.open(player, pipe.getX(), pipe.getY(), pipe.getZ());
+    }
+
+    public static IKey reallyDynamicKey(Supplier<IKey> keySupplier) {
```

Comment:
You can use `DynamicDrawable` since `overlay` accepts `IDrawable`, not just `IKey`

Reply:
Oh right, that's why you requested dynamic `LangKey`. Now you can use it 😉

Reply:
Oh I meant "it" by new supplier for `LangKey`, not `reallyDynamicKey`... English is hard...